File: cpp/0001.两数之和.cpp
class Solution {
public:
    // [题解]1. O(n) t:8ms(92%) O(n) m:10.4M(50%)
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashtable;
        for (int i = 0; i < nums.size(); ++i) {
            auto it = hashtable.find(target - nums[i]);
            if (it != hashtable.end()) return {it->second, i};
            hashtable[nums[i]] = i;
        }
        return {};
    }
};

File: java/0001.两数之和.java
class Solution {
    // 1. O(n) t:1ms(100%) m:39.8M(19%) Hash
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++){
            int x = nums[i];
            if (map.containsKey(target - x)){
                return new int[]{map.get(target - x), i};
            }
            map.put(x, i);
        }
        throw new IllegalArgumentException("No two sum solution");
    }
}
